hamster_lib/objects.py

```diff
@@ -399,83 +394,33 @@ def create_from_raw_fact(cls, raw_fact, config=None):
-                    most left ``@`` symbol. If no such symbol was present at all,
-                    ``back=None``. Both substrings have been trimmed of any leading
-                    and tailing whitespace.
-
-            Note:
-                If our string contains multiple ``@`` symbols, all but the most left
-                one will be treated as part of the regular ``back`` string.
-                This allows for usage of the symbol in descriptions, categories and tags.
-            """
-            result = string.split('@', 1)
-            length = len(result)
-            if length == 1:
-                front, back = result[0].strip(), None
-            else:
-                front, back = result
-                front, back = front.strip(), back.strip()
-            return (front, back)
-
-        def comma_split(string):
-            """
-            Split string at the most left comma.
-
-            Args:
-                string (str): String to be processed. At this stage this should
-                    look something like ``<Category>, <Description>
-
-
-            Returns
-                tuple: (category_and_tags, description). Both substrings have their
-                    leading/tailing whitespace removed.
-                    ``category_and_tags`` may include >=0 tags indicated by a leading ``#``.
-                    As we have used the most left ``,`` to separate both substrings that
-                    means that categories and tags can not contain any ``,`` but the
-                    description text may contain as many as wished.
-            """
-
-            result = tuple(string.split(',', 1))
-            length = len(result)
-            if length == 1:
-                category, description = result[0].strip(), None
-            else:
-                category, description = tuple(result)
-                category, description = category.strip(), description.strip()
-            return (category.strip(), description)
+        Raises:
+            ValueError: If we fail to extract at least ``start`` or ``activity.name``.
+            ValueError: If ``end <= start``.
+
+        """
 
         if not config:
             config = {'day_start': datetime.time(0, 0, 0)}
 
-        time_info, rest = time_helpers.extract_time_info(raw_fact)
-        start, end = time_helpers.complete_timeframe(time_info, config, partial=True)
-        activity_name, back = at_split(rest)
+        extracted_components = parse_raw_fact(raw_fact)
 
-        if back:
-            category_name, description = comma_split(back)
-            if category_name:
-                category = Category(category_name)
-            else:
-                category = None
+        start, end = time_helpers.complete_timeframe(extracted_components['timeinfo'],
+            config, partial=True)
+        start, end = time_helpers.validate_start_end_range((start, end))
+
+        activity_name = extracted_components.get('activity')
```

Comment:
You are right, in a way. What ``dict.get()`` ensures is that we always get a value even if the *key* is not in the dict. Now this clearly will not happen right now, but I find myself using it ever more often to make dict lookups less error prone.
Then again, failing loudly if the key is not present would be preferable here, as it would indicate that something went *really* wrong. Will fix accordingly.